### PR TITLE
Halborn/angular audit fix

### PR DESCRIPTION
**Pull request by mikolaichuk**
## Issue
https://app.clickup.com/t/2rbudp3

## Description
Fixes of Halborn findings. 
Should be merged in halborn/angular-audit.

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Review by comfi01 — APPROVED** [review]

**Review by kevin-composable — APPROVED** [review]
Thanks for moving the changes into this branch

File: frame/lending/src/lib.rs [hunk 1]
```diff
@@ -1249,102 +1351,14 @@ pub mod pallet {
...
-
-			<T as Config>::MultiCurrency::transfer(
-				market.collateral_asset,
-				account,
-				&market_account,
-				amount,
-				true,
-			)?;
-			Ok(())
+			Self::do_deposit_collateral(market_id, account, amount.try_into_validated()?)
```
Comment (dzmitry-lahoda) [inline frame/lending/src/lib.rs:1354]: idea of validated it validates during serde, so sure about this applicane. so may be ok

File: frame/lending/src/lib.rs [hunk 2]
```diff
@@ -1129,6 +1129,108 @@ pub mod pallet {
...
+				.unwrap_or_else(|_| CollateralLpAmountOf::<Self>::zero());
+
+			ensure!(amount <= collateral_balance, Error::<T>::NotEnoughCollateralToWithdraw);
+
+			let borrow_asset = T::Vault::asset_id(&market.borrow_asset_vault)?;
+			let borrower_balance_with_interest =
+				Self::total_debt_with_interest(market_id, account)?.unwrap_or_zero();
+
+			let borrow_balance_value =
+				Self::get_price(borrow_asset, borrower_balance_with_interest)?;
```
Comment (dzmitry-lahoda) [inline frame/lending/src/lib.rs:1152]: assume that copied as is mostly, looks like so

File: frame/liquidations/src/lib.rs [hunk 3]
```diff
@@ -144,7 +147,7 @@ pub mod pallet {
 			Ok(().into())
 		}
 
-		#[pallet::weight(T::WeightInfo::sell())]
+		#[pallet::weight(T::WeightInfo::sell(if configuration.is_empty() { 1_u32 } else { configuration.len() as u32 }))]
```
Comment (dzmitry-lahoda) [inline frame/liquidations/src/lib.rs:150]: configuraiton.len().max(1) as u32

**Review by dzmitry-lahoda — APPROVED** [review]

**Pull request merged**